src/sparsezoo/v2/model_objects.py

```diff
@@ -0,0 +1,250 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Class objects for standardization and validation of a model folder structure
+"""
+
+
+import logging
+import re
+from collections import OrderedDict
+from typing import List, Optional
+
+import onnx
+
+from sparsezoo.utils.numpy import load_numpy_list
+from sparsezoo.v2.directory import Directory
+from sparsezoo.v2.file import File
+
+
+__all__ = ["FrameworkFiles", "NumpyDirectory", "SampleOriginals"]
+
+NUMPY_DIRECTORY_NAMES = ["sample-inputs", "sample-outputs"]
+
+
+class FrameworkFiles(Directory):
+    """
+    Object that represents a directory with framework files.
+
+    :param files: list of files contained within the FrameworkFiles
+    :param name: name of the FrameworkFiles
+    :param path: path of the FrameworkFiles
+    :param url: url of the FrameworkFiles
+    """
+
+    def __init__(
+        self,
+        files: List[File],
+        name: str,
+        path: Optional[str] = None,
+        url: Optional[str] = None,
+    ):
+        super().__init__(files=files, name=name, path=path, url=url)
+
+        self.valid_nested_folder_patterns = ["checkpoint_(.*)/", "logs/"]
+
+    # TODO: Add support for model cards (pull integration from
+    #  model.md and specify which files to validate)
+    def validate(self, integration: Optional[str] = None) -> bool:
+        """
+        Validates the structure of framework files.
+
+        :param integration: integration: name of the specific integration
+            (e.g. transformers, YOLOv5 etc.)
+        :return: boolean flag; True if files are valid and no errors arise
+        """
+        validations = {}
+        for file in self.files:
+            if isinstance(file, Directory):
+                # check for Directory files
+                validations[file.name] = self._check_directory(directory=file)
+            else:
+                # check for File files
+                validations[file.name] = self._check_file(file=file)
+
+        if not all(validations.values()):
+            logging.warning(
+                f"Following files: "
+                f"{[key for key, value in validations.items() if not value]} "
+                "were not successfully validated."
+            )
+
+        return all(validations.values())
```

Comment:
Super bad nit: Can be ignored

Re-computation of result, could we store `all(validation.values())` in a variable and return that

Reply:
Haha, this is absolutely true. However, given that `validation` is a `dict`, it will not gives us significant computational overhead. I will keep the current version of the code, as I think it gives better readability.